September v3 Release Notes (#7101)
adobe/react-spectrum: packages/dev/docs/pages/releases/2024-09-30.mdx

of the License at http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License. */}

import {BlogPostLayout, Hero} from '@react-spectrum/docs';
export default BlogPostLayout;

---

description: Happy fall! We’re excited to announce that our color picker components are now generally available! This includes ColorPicker, ColorArea, ColorField, ColorSlider, ColorSwatch, ColorSwatchPicker, and ColorWheel. These enable you to build fully customizable color pickers, including accessible color descriptions for screen reader support.
In addition, we’ve introduced a new pending state for buttons in React Aria Components, allowing you to display a progress spinner to indicate that an action is in progress. It also handles screen reader announcements and disables other interactions, except for focus, during the pending state.

> We're also advancing the React Aria Components Tree and Spectrum TreeView to beta, and have added some basic documentation. In alpha, we have new Disclosure and DisclosureGroup components in React Aria Components, which can be used to build accordions and other collapsible panels.

Reviewer: Happy fall! 🍁 We’re excited to announce that our color picker components are now generally available! We've also added a pending state for buttons in React Aria Components, advanced TreeView to beta, and added new Disclosure and DisclosureGroup components in alpha for building accordions and collapsible panels.

date: 2024-09-30
---
# September 30, 2024 Release
Happy fall! 🍁 We’re excited to announce that our color picker components are now generally available! This includes ColorPicker, ColorArea, ColorField, ColorSlider, ColorSwatch, ColorSwatchPicker, and ColorWheel. These enable you to build fully customizable color pickers, including accessible color descriptions for screen reader support.

> In addition, we’ve introduced a new pending state for buttons in React Aria Components, allowing you to display a progress spinner to indicate that an action is in progress. It also handles screen reader announcements and disables other interactions, except for focus, during the pending state.

Reviewer: Pretty sure the paragraphs also need space between them

{/* Copyright 2024 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License. */}

import {BlogPostLayout, Hero} from '@react-spectrum/docs';
export default BlogPostLayout;

---

description: Happy fall! 🍁 We’re excited to announce that our color picker components are now generally available! We've also added a pending state for buttons in React Aria Components, advanced TreeView to beta, and added new Disclosure and DisclosureGroup components in alpha for building accordions and collapsible panels.

date: 2024-09-30
---

# September 30, 2024 Release

> Happy fall! 🍁 We’re excited to announce that our color picker components are now generally available! This includes [ColorPicker](https://react-spectrum.adobe.com/react-aria/ColorPicker.html), [ColorArea](https://react-spectrum.adobe.com/react-aria/ColorArea.html), [ColorField](https://react-spectrum.adobe.com/react-aria/ColorField.html), [ColorSlider](https://react-spectrum.adobe.com/react-aria/ColorSlider.html), [ColorSwatch](https://react-spectrum.adobe.com/react-aria/ColorSwatch.html), [ColorSwatchPicker](https://react-spectrum.adobe.com/react-aria/ColorSwatchPicker.html), and [ColorWheel](https://react-spectrum.adobe.com/react-aria/ColorWheel.html). These enable you to build fully customizable color pickers, including accessible color descriptions for screen reader support.

Reviewer 2: ```suggestion
Happy fall! 🍁 We’re excited to announce that our color picker components are now generally available! This includes [ColorPicker](https://react-spectrum.adobe.com/react-aria/ColorPicker.html), [ColorArea](https://react-spectrum.adobe.com/react-aria/ColorArea.html), [ColorField](https://react-spectrum.adobe.com/react-aria/ColorField.html), [ColorSlider](https://react-spectrum.adobe.com/react-aria/ColorSlider.html), [ColorSwatch](https://react-spectrum.adobe.com/react-aria/ColorSwatch.html), [ColorSwatchPicker](https://react-spectrum.adobe.com/react-aria/ColorSwatchPicker.html), and [ColorWheel](https://react-spectrum.adobe.com/react-aria/ColorWheel.html). These enable you to build fully customizable color pickers, and include accessible color descriptions for screen reader support.
```

Reviewer: they don't allow you to include accessible descriptions, they do include them. 😄 maybe "which include" 🤷

Reviewer 2: How about:

These components include accessible color descriptions for screen reader support and enable you to build fully customizable color pickers.

In addition, we’ve introduced a new [pending state](https://react-spectrum.adobe.com/react-aria/Button.html#pending) for buttons in React Aria Components, allowing you to display a progress spinner to indicate that an action is in progress. It also handles screen reader announcements and disables other interactions, except for focus, during the pending state.

> We're also advancing the React Aria Components [Tree](https://react-spectrum.adobe.com/react-aria/Tree.html) and Spectrum [TreeView](https://react-spectrum.adobe.com/react-spectrum/TreeView.html) to beta, and have added some basic documentation. In alpha, we have new [Disclosure](https://react-spectrum.adobe.com/react-aria/Disclosure.html) and [DisclosureGroup](https://react-spectrum.adobe.com/react-aria/DisclosureGroup.html) components in React Aria Components, which can be used to build accordions and other collapsible panels.

Reviewer 2: ```suggestion
React Aria Components [Tree](https://react-spectrum.adobe.com/react-aria/Tree.html) and Spectrum [TreeView](https://react-spectrum.adobe.com/react-spectrum/TreeView.html) are moving to beta and we've started their documentation. We are introducing the new [Disclosure](https://react-spectrum.adobe.com/react-aria/Disclosure.html) and [DisclosureGroup](https://react-spectrum.adobe.com/react-aria/DisclosureGroup.html) components in React Aria Components. These can be used to build accordions and other collapsible panels.
```

Reviewer: I prefer the original 🤷

Reviewer 2: I don't have a strong preference. I was mostly trying to change "In alpha, we have" which sounded too different from "We're also advancing". The sentence structures just don't match up for me.
